### PR TITLE
Use for comprehension to find command

### PR DESCRIPTION
**Pull request by ashwinbhaskar**

### CODE REVIEW

File: src/main/scala/com/pharway/ting/CommandLine.scala [hunk 1]
```diff
@@ -47,21 +49,29 @@ object CommandLine
...
+          case Success(cmd) =>
+            if cmd.verb != Init && cmd.subject != Project
+              if(ProjectOps.validProject())
+                cmd.func(options) 
+              else
+                CommandError("Not a valid project.")
+            else
+              cmd.func(options)
+          case Failure(_) => helpText
+      case _ => helpText
```
Comment (olofwalker) [inline src/main/scala/com/pharway/ting/CommandLine.scala]: Missing EOL

File: src/main/scala/com/pharway/ting/CommandLine.scala [hunk 2]
```diff
@@ -47,21 +49,29 @@ object CommandLine
       commands.map(cmd => 
         f"${cmd.syntax}%-45s - ${cmd.description}"
         ).mkString("\n")
-  
+
+    def findCommand(verb: Verb, subject: Subject): Try[Command] = 
+      commands.find(_.compare(verb, subject)) match
+        case Some(command) => Success(command)
+        case None => Failure(Exception("Not a valid command"))
```
Comment (olofwalker) [inline src/main/scala/com/pharway/ting/CommandLine.scala]: Now when the information is in hand, we could add what command was not valid here. Something like:

`Exception(s"'$verb $subject' is not a recognised command")`

File: src/main/scala/com/pharway/ting/CommandLine.scala [hunk 3]
```diff
@@ -47,21 +49,29 @@ object CommandLine
...
+          yield 
+            command  
+          
+        commandOrError match 
+          case Success(cmd) =>
+            if cmd.verb != Init && cmd.subject != Project
+              if(ProjectOps.validProject())
+                cmd.func(options) 
+              else
+                CommandError("Not a valid project.")
```
Comment (olofwalker) [inline src/main/scala/com/pharway/ting/CommandLine.scala]:
```suggestion
                CommandError("The current folder does not contain a valid project. You can initialize a project in this folder using the command `ting init project`.")
```

File: src/main/scala/com/pharway/ting/CommandLine.scala [hunk 4]
```diff
@@ -39,6 +39,8 @@ object CommandLine
     Command(Init,Project,"init project","Initialize a new project in the current folder.",ProjectOps.init)
   )
 
+  def synthesize(verb: String, subject: String): Try[(Verb, Subject)] =
```
[the remark this answers is not in the document]
Comment (ashwinbhaskar) [reply src/main/scala/com/pharway/ting/CommandLine.scala]: Done

File: src/main/scala/com/pharway/ting/CommandLine.scala (hunk 2 above)
Comment (ashwinbhaskar) [reply src/main/scala/com/pharway/ting/CommandLine.scala]: Done

File: src/main/scala/com/pharway/ting/CommandLine.scala (hunk 3 above)
Comment (ashwinbhaskar) [reply src/main/scala/com/pharway/ting/CommandLine.scala]: Done

File: src/main/scala/com/pharway/ting/CommandLine.scala (hunk 1 above)
Comment (ashwinbhaskar) [reply src/main/scala/com/pharway/ting/CommandLine.scala]: Done

**Comment by ashwinbhaskar** [comment]
@olofwalker hey, I have made the changes and pushed.

**Pull request merged**